### PR TITLE
feat(terminal): support selecting different profile

### PR DESCRIPTION
**Pull request by elianiva**
### Related GitHub Issue

Closes: #7892

### Description

This PR adds support for setting preferred profile for the terminal spawned by RooCode. The items are fetched from VSCode's setting so any user added profiles will be present.

### Pre-Submission Checklist

- [x] **Issue Linked**: This PR is linked to an approved GitHub Issue (see "Related GitHub Issue" above).
- [x] **Scope**: My changes are focused on the linked issue (one major feature/fix per PR).
- [x] **Self-Review**: I have performed a thorough self-review of my code.
- [x] **Testing**: New and/or updated tests have been added to cover my changes (if applicable).
- [x] **Documentation Impact**: I have considered if my changes require documentation updates (see "Documentation Updates" section below).
- [x] **Contribution Guidelines**: I have read and agree to the [Contributor Guidelines](/CONTRIBUTING.md).

### Screenshots / Videos

image

### Get in Touch

@elianiva

----

> [!IMPORTANT]
> Adds support for selecting a preferred terminal profile in RooCode, integrating with VSCode settings and updating terminal creation logic.
> 
>   - **Behavior**:
>     - Adds `terminalPreferredProfile` setting in `global-settings.ts` to select a preferred terminal profile.
>     - Integrates with VSCode terminal profiles, allowing users to choose a profile or use the default.
>     - Updates `ClineProvider.ts` and `Terminal.ts` to apply the selected profile when creating terminals.
>   - **Services**:
>     - Introduces `TerminalProfileService` to manage terminal profiles, including fetching available profiles and configurations.
>     - Provides methods to get default and specific profiles, and to configure terminal options based on profiles.
>   - **UI**:
>     - Updates `SettingsView.tsx` and `TerminalSettings.tsx` to include a dropdown for selecting terminal profiles.
>     - Adds localization support for new settings in multiple language files.
>   - **Tests**:
>     - Adds `TerminalProfileService.spec.ts` to test profile fetching and configuration logic.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for b3483335825e61ecd836cfa36f45c911c67e8712. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: webview-ui/src/components/settings/TerminalSettings.tsx [hunk 1]
```diff
@@ -199,6 +218,45 @@ export const TerminalSettings = ({
...
+								onValueChange={(value) => {
+									const profileValue = value === "default" ? undefined : value
+									setCachedStateField("terminalPreferredProfile", profileValue)
+								}}
+								data-testid="terminal-preferred-profile-dropdown">
+								<SelectTrigger className="w-full">
+									<SelectValue placeholder="Select profile..." />
+								</SelectTrigger>
+								<SelectContent>
+									<SelectItem value="default">Default (System)</SelectItem>
```
[the remark this answers is not in the document]
Comment (elianiva) [reply webview-ui/src/components/settings/TerminalSettings.tsx:236]: can't do that, select item requires non-empty string

**Comment by ghost** [comment]
[See this task on Roo Code Cloud](https://app.roocode.com/cloud-jobs/XSfHTblD?utm_source=cloud-job-url&utm_medium=link&utm_campaign=github.pr.review.sync)

Re-review completed. The latest commit only restores Korean translations lost during merge resolution. No new issues found. The 3 previously identified issues remain unresolved:

- [ ] Fix misleading comment in Terminal.ts about conditional name/icon override
- [ ] Add test coverage for empty string handling in TerminalProfileService
- [ ] Align default option label between frontend ("Default (System)") and backend ("Default (VSCode Default)")

<details>
<summary>Previous Reviews</summary>

- e50523f: [Review #1](https://app.roocode.com/cloud-jobs/JiBuLs9H?utm_source=cloud-job-url&utm_medium=link&utm_campaign=previous-reviews)
</details>


<sub>Mention @roomote in a comment to trigger your PR Fixer agent and make changes to this pull request.</sub>

File: src/integrations/terminal/__tests__/TerminalProfileService.spec.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+			}
+			mockGetConfiguration.mockReturnValue(mockConfig as any)
+
+			const options = TerminalProfileService.getTerminalOptionsForRoo()
+			expect(options).toEqual({
+				shellPath: "/bin/bash",
+				shellArgs: ["--login"],
+			})
+		})
+	})
```
Comment (ghost) [inline src/integrations/terminal/__tests__/TerminalProfileService.spec.ts:198]: [P4] Missing test coverage for empty string equivalence. The code treats empty string and undefined as equivalent (line 150: `if (preferredProfile)` returns false for both "", null, and undefined) but no test verifies this critical sentinel behavior. Add a test case: `getTerminalOptionsForRoo("")` should return undefined (using VSCode default), not fall back to default profile.

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=RooCodeInc/Roo-Code&prNumber=8487&fixId=0c988237ad) or mention @roomote and request a fix.</sub>

**Comment by daniel-lxs** [comment]
Thanks for working on this terminal profile feature! After testing it out with the team, we found a few issues that make it tricky to use in practice.

The main problem is that it shows all possible profiles even when they're not actually installed, and if you pick one that doesn't exist, commands just fail. Also since we switched to inline terminal as the default, it doesn't really work with that setup anymore.

Ideally something like this would only show profiles that are actually available, gracefully fall back when a profile goes missing, and work well with both inline and regular terminals. The current approach has some rough edges that make it not quite ready.

I'm going to close this for now, but definitely let me know if you want to take another shot at it with those things in mind. Happy to help brainstorm or review a different approach!
